agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -270,6 +274,92 @@ private static void overlayConfiguration(
       String hostname = HostName.get();
       config.role.instance = hostname == null ? "unknown" : hostname;
     }
+    supportTelemetryProcessorsOldSemConv(config);
+  }
+
+  private static void supportSamplingOverridesOldSemConv(SamplingOverride override) {
+    for (Configuration.SamplingOverrideAttribute attribute : override.attributes) {
+      attribute.key = mapAttributeKey(attribute.key);
+    }
+  }
+
+  private static void supportTelemetryProcessorsOldSemConv(Configuration config) {
```

Comment:
can we emit a warning when we find that the user is using one of the old attributes?